File: src/main/java/com/indiepost/service/PostService.java
package com.indiepost.service;

import com.indiepost.dto.FullTextSearchQuery;
import com.indiepost.dto.post.PostDto;
import com.indiepost.dto.post.PostQuery;
import com.indiepost.dto.post.PostSummaryDto;
import com.indiepost.dto.post.RelatedPostResponseDto;

import java.time.LocalDateTime;
import java.util.List;

/**
 * Created by jake on 7/30/16.
 */
public interface PostService {

    PostDto findOne(Long id);

    PostDto findOneByLegacyId(Long id);

    Long count();

    Long count(PostQuery query);

    List<PostSummaryDto> findByIds(List<Long> ids);

    List<PostSummaryDto> find(int page, int maxResults, boolean isDesc);

    List<PostSummaryDto> findByCategoryId(Long categoryId, int page, int maxResults, boolean isDesc);

    List<PostSummaryDto> findByTagName(String tagName, int page, int maxResults, boolean isDesc);

    List<RelatedPostResponseDto> findRelatedPosts(List<Long> ids, boolean isLegacy, boolean isMobile);

    List<PostSummaryDto> findTopRatedPosts(LocalDateTime since, LocalDateTime until, Long limit);

    List<PostSummaryDto> findScheduledPosts();

    List<PostSummaryDto> search(PostQuery query, int page, int maxResults, boolean isDesc);

    List<PostSummaryDto> fullTextSearch(FullTextSearchQuery query);

    PostSummary findSplashPost();

    PostSummaryDto findFeaturePost();

    List<PostSummaryDto> findPickedPosts();
}

File: src/main/kotlin/com/indiepost/service/PostService.kt
package com.indiepost.service

import com.indiepost.dto.PageWithProfile
import com.indiepost.dto.post.PostDto
import com.indiepost.dto.post.PostQuery
import com.indiepost.dto.post.PostSummaryDto
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.time.LocalDateTime

/**
 * Created by jake on 7/30/16.
 */
interface PostService {

    fun findOne(postId: Long): PostDto

    fun count(): Long

    fun count(postQuery: PostQuery): Long

    fun findPublicPosts(pageable: Pageable): Page<PostSummaryDto>

    fun findByTagName(tagName: String, pageable: Pageable): Page<PostSummaryDto>

    fun query(postQuery: PostQuery, pageable: Pageable): Page<PostSummaryDto>

    fun fullTextSearch(text: String, pageable: Pageable): Page<PostSummaryDto>

    fun findRelatedPostsById(id: Long, pageable: Pageable): Page<PostSummaryDto>

    fun recommendations(pageable: Pageable): Page<PostSummaryDto>

    fun findReadingHistory(pageable: Pageable): Page<PostSummaryDto>

    fun findBookmarks(pageable: Pageable): Page<PostSummaryDto>

    fun findByIds(ids: List<Long>): List<PostSummaryDto>

    fun findTopRatedPosts(since: LocalDateTime, until: LocalDateTime, limit: Int): List<PostSummaryDto>

    fun findScheduledPosts(): List<PostSummaryDto>

    fun findPickedPosts(): List<PostSummaryDto>

    fun findSplashPost(): PostSummaryDto?

    fun findFeaturePost(): PostSummaryDto?

    fun defaultRecommendations(pageable: Pageable): Page<PostSummaryDto>

    fun findByProfileSlug(slug: String, pageable: Pageable): PageWithProfile<PostSummaryDto>
}